stories/src/main/java/com/wordpress/stories/compose/text/FontRequester.kt

```diff
@@ -0,0 +1,72 @@
+package com.wordpress.stories.compose.text
+
+import android.content.Context
+import android.content.res.Resources.NotFoundException
+import android.graphics.Typeface
+import android.util.Log
+import androidx.annotation.FontRes
+import androidx.core.content.res.ResourcesCompat
+import androidx.core.content.res.ResourcesCompat.FontCallback
+import androidx.core.provider.FontsContractCompat.FontRequestCallback.FontRequestFailReason
+
+/**
+ * Handles dispatching requests to ResourcesCompat.getFont() asynchronously. These requests can take a long time,
+ * for example when the font is downloadable and there are connection issues.
+ *
+ * When registering or requesting a font from FontRequester, a fallback Typeface must be specified which will
+ * be returned if:
+ * - The request to ResourcesCompat hasn't completed yet
+ * - The request to ResourcesCompat failed for any reason
+ * - The font being requested wasn't previously registered with FontRequester
+ */
+object FontRequester {
+    private val TAG = FontRequester::class.java.simpleName
+
+    val fontMap = mutableMapOf<Int, Typeface>()
```

Comment:
I've seen Kotlin provides a `[LinkedHashMap](https://docs.oracle.com/javase/8/docs/api/java/util/LinkedHashMap.html)` as the implementation of `mutableMapOf`.
A LinkedHashMap is not thread safe, so I wondered what would happen in the (possibly very rare) case where several fonts are retrieved and then their async callbacks decide to assign values to it (write) at the same time.

Generally I'd think there wouldn't be a problem given all keys in the map are guaranteed to be different, but it really depends on the implementation (maybe registering a new key makes values to shift in the underlying data structure, which would then potentially be problematic given the insertion method is not synchronized).

So in order to avoid this, this would need to be externally synchronized, one way is to use `Collections.synchronizedMap`  wdyt?

```suggestion
    val fontMap = Collections.synchronizedMap(mutableMapOf<Int, Typeface>())
```

Reply:
Good point @mzorz , let's be safe 👍 Made the change in 5e79486 (the `fontMap` type needs an explicit cast for null safety reasons, otherwise it generates a warning).